core/treepath/treepath.go

```diff
@@ -0,0 +1,257 @@
+// Package treepath provides functionality for manipulating treepaths.
+//
+// Treepaths are string-encodings of paths through trees.
+//
+// Treepaths come from
+// https://github.com/Kashomon/glift-core/blob/master/src/rules/treepath.js
+package treepath
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/otrego/clamshell/core/game"
+)
+
+// A Treepath is a list of variations that says how to travel through a tree of
+// moves. And has two forms a list version and astring version. First, the
+// list-version:
+//
+//    [0,1,0]
+//
+// Means we will first take the 0th variation, then we will take the 1ist
+// variation, and lastly we will take the 0th variation again. For
+// convenience, the treepath can also be specified by a string, which is where
+// the fun begins. At it's simpliest,
+//
+//    [0,0,0] becomes 0.0.0
+//
+// but there are a couple different string short-hands that make using treepaths
+// a little easier
+//
+//    0.1:2   Take the 0th variation, then repeat taking the 1st varation twice
+//
+// There are two types of treepaths discussed below -- A *treepath fragment*
+// (which is what we have been describing) and an *initial treepath*.
```

Comment:
Can this description give a sense of where each would be used?

It seems like the advantage of initial treepaths is that they're slightly shorter, but IIUC only saving 2 characters while adding some conceptual overhead: `0:53` vs. `53`. Is there more to it than that?

Reply:
Awesome. Love it. This is a great simplification.

Reply:
Should it not say there are two types anymore?

Still useful to note the idiom for skipping a series of initial moves if it's especially common, it's just not a standalone "type" anymore.

Reply:
Done